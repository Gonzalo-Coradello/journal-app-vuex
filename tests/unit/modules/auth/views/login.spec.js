import Login from '@/modules/auth/views/Login.vue'
import { shallowMount } from '@vue/test-utils'
import { describe, expect, test, vi } from 'vitest'
import createVuexStore from '../../../mock-data/mock-store'
import Swal from 'sweetalert2'

describe('Login component', () => {
  const store = createVuexStore({
    status: 'not-authenticated',
    user: null,
    idToken: null,
    refreshToken: null
  })

  store.dispatch = vi.fn()

  test('debe hacer match con el snapshot', () => {
    const wrapper = shallowMount(Login, {
      global: {
        plugins: [store]
      }
    })

    expect(wrapper.html()).toMatchSnapshot()
  })

  test('debe disparar el SWAL cuando las credenciales son incorrectas', async () => {
    store.dispatch.mockReturnValueOnce({ ok: false, message: 'Error en credenciales' })

    const wrapper = shallowMount(Login, {
      global: {
        plugins: [store]
      }
    })

    vi.spyOn(Swal, 'fire')

    await wrapper.find('form').trigger('submit')

    expect(store.dispatch).toHaveBeenCalledWith('auth/signInUser', { email: '', password: '' })
    expect(Swal.fire).toHaveBeenCalledWith('Error', 'Error en credenciales', 'error')
  })

  test('debe redirigir a la ruta no-entry con credenciales correctas', async () => {
    store.dispatch.mockReturnValueOnce({ ok: true })

    const wrapper = shallowMount(Login, {
      global: {
        plugins: [store]
      }
    })

    const [txtEmail, txtPassword] = wrapper.findAll('input')

    await txtEmail.setValue('test@test.com')
    await txtPassword.setValue('123456')

    await wrapper.find('form').trigger('submit')

    expect(store.dispatch).toHaveBeenCalledWith('auth/signInUser', {
      email: 'test@test.com',
      password: '123456'
    })

    expect(wrapper.router.push).toHaveBeenLastCalledWith({ name: 'no-entry' })
  })
})
