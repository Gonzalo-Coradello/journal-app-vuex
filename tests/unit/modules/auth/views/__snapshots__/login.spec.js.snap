// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Login component > debe hacer match con el snapshot 1`] = `
"<span class="login100-form-title p-b-41"> Ingresar </span>
<form class="login100-form validate-form p-b-33 p-t-5">
  <div class="wrap-input100 validate-input" data-validate="Igresar email"><input class="input100" type="text" placeholder="Correo" required=""><span class="focus-input100" data-placeholder=""></span></div>
  <div class="wrap-input100 validate-input" data-validate="Ingresar contraseña"><input class="input100" type="password" placeholder="Contraseña" required=""><span class="focus-input100" data-placeholder=""></span></div>
  <div class="container-login100-form-btn m-t-32"><button type="submit" class="login100-form-btn">Login</button></div>
  <div class="container-login100-form-btn m-t-32">
    <router-link-stub to="[object Object]" replace="false" custom="false" ariacurrentvalue="page"></router-link-stub>
  </div>
</form>"
`;
